lib/main.js

```diff
@@ -385,54 +392,192 @@ function createPipelinedRequest(cb){
  * @param options Options for the request function
  * @param cb  Callback to handle response
  * */
-function request(options,cb){
-    var _options={};
-    if(isThrottled()){
-        var host=options.host;
-        var port=options.port;
-        var path=options.path;
-        var method=options.method;
-        var proxyPort=process.pid%60000;
-        _options.host=localProxyHost;
-        _options.port=proxyPort;
-        _options.method=method;
-        if(options.headers !=null){
-            _options.headers=options.headers;
-        }else{
-            _options.headers={};
+function request(options, cb) {
+    var _options = {};
+    if (isThrottled()) {
+        var host = options.host;
+        var port = options.port;
+        var path = options.path;
+        var method = options.method;
+        var proxyPort = process.pid % 60000;
+        _options.host = localProxyHost;
+        _options.port = proxyPort;
+        _options.method = method;
+        if (options.headers != null) {
+            _options.headers = options.headers;
+        } else {
+            _options.headers = {};
         }
-        _options.headers['host']=host;
-        _options.headers['port']=port;
-        _options.headers['path']=path;
-        _options.headers['method']=method;
-    }else{
-        _options=options;
+        _options.headers['host'] = host;
+        _options.headers['port'] = port;
+        _options.headers['path'] = path;
+        _options.headers['method'] = method;
+    } else {
+        _options = options;
+    }
+//--------------------------Anonymizating Request----------------------------------------------------------------------------------
```

Comment:
All this code of anonymizing request should not be added under main.js. You have to keep the changes in main.lib as minimum as possible and pass the responsibility to the anonymizer lib.

Check my previous comment on the anonymizationEnabled variable, and move all this new code under anonymizer.js with proper refactoring.

I would expect something like this:
```
if(anonymizer.isOn){
  anonymizer.startProcess(options, ..);
}
```

Reply:
Additionally, you are using excessively console.log. Are you able to add breakpoints and debug properly through your IDE? Which IDE are you using?

Remove all those console.log, and keep them only for informative purposes that give value to a developer

Reply:
For the IDE, I am using VS Code, I used debugging feature for other IDEs before but never tried to use it at VS Code with javascript, I will adapt to use it, thank you for point out.